lib/components/ApproveReject.js

```diff
@@ -96,14 +109,23 @@ class ApproveReject extends PureComponent {
   renderProposalInfo() {
     let { proposalInfo, proposalValue, proposalRecipient } = this.props;
 
-    const { cosignerToken, selectedHardware, accountNumber } = this.state;
+    const {
+      cosignerToken,
+      selectedHardware,
+      accountNumber,
+      advanced,
+      derivationPath,
+    } = this.state;
 
     // add properties that may be missing
     proposalInfo = this.makeProposalSafe(proposalInfo);
 
+    // TODO: unHACK the copy/paste
```

Comment:
There are now many views that use the same logic for `advanced` mode that allows for a path to be provided.
The HD Path component would be great to simplify this so that we can use the same component in many places
and allow for more flexible UX